src/main/java/tutorlink/command/AddGradeCommand.java

```diff
@@ -30,43 +30,17 @@ public CommandResult execute(AppState appstate, HashMap<String, String> hashmap)
             throw new IllegalValueException(Commons.ERROR_NULL);
         }
 
-        //Get component object using String componentDescription
-        ComponentList componentFilteredList = appstate.components.findComponent(componentDescription.toUpperCase());
-        Component component;
-        if (componentFilteredList.size() == 1) {
-            component = componentFilteredList.getComponentArrayList().get(0);
-        } else if (componentFilteredList.size() == 0) {
-            throw new ComponentNotFoundException(String.format(Commons.ERROR_COMPONENT_NOT_FOUND,
-                    componentDescription));
-        } else {
-            String errorMessage = String.format(Commons.ERROR_DUPLICATE_COMPONENT, componentDescription);
-            throw new DuplicateComponentException(errorMessage);
-        }
+        Component component = findComponentFromComponents(appstate, componentDescription);
```

Comment:
I like that the code for Find and ConvertScore have been extracted out into individual methods.